integration/tests/cook/test_basic.py

```diff
@@ -1859,6 +1859,18 @@ def test_basic_docker_job(self):
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertIn('success', [i['status'] for i in job['instances']])
 
+    @unittest.skipUnless(util.docker_tests_enabled(), "Requires a test docker image")
+    def test_default_container_for_pool(self):
+        # Special logic in util.submit_jobs.full_spec maps container=None and removes it from the submitted job spec.
+        job_uuid, resp = util.submit_job(
+            self.cook_url,
+            command='cat /.dockerenv',
+            container=None,
+            max_retries=5)
+        self.assertEqual(resp.status_code, 201)
+        job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
+        self.assertIn('success', [i['status'] for i in job['instances']])
```

Comment:
Should we also assert that the container the job ended up getting was the configured one?

Reply:
Done

Reply:
Since we're defaulting more than just the image, should we assert on everything we're defaulting being correct?

Reply:
Which other fields should we check? Image is the only one we set by default in the integration test suite (via our ENV variable.)

Reply:
Oh I thought we were defaulting more than just that, e.g.:

```
                                           {:type "docker"
                                            :docker {
                                                     :image "python:3.6",
                                                     :network "HOST"
                                                     :force-pull-image false
                                                     :parameters []}}
```

Reply:
I pulled that out of what the integration tests normally submit, except that integration tests submit a different image name (python:3.5.9-stretch) that works with minimesos.

Reply:
Sounds good